spec/services/flow/steps/partner/cash_outgoings_step_spec.rb

```diff
@@ -0,0 +1,69 @@
+require "rails_helper"
+
+RSpec.describe Flow::Steps::Partner::CashOutgoingsStep, type: :request do
+  let(:legal_aid_application) { create(:legal_aid_application) }
+
+  before do
+    allow(legal_aid_application).to receive(:housing_payments_for?).with("Applicant").and_return(true)
+    allow(legal_aid_application).to receive(:housing_payments_for?).with("Partner").and_return(false)
+    allow(legal_aid_application).to receive(:uploading_bank_statements?).and_return(true)
+  end
+
+  describe "#path" do
+    subject { described_class.path.call(legal_aid_application) }
+
+    it { is_expected.to eql providers_legal_aid_application_partners_cash_outgoing_path(legal_aid_application) }
+  end
+
+  describe "#forward" do
+    subject { described_class.forward.call(legal_aid_application) }
+
+    context "when the applicant has housing payments and is uploading bank statements" do
+      it { is_expected.to be :housing_benefits }
+    end
+
+    context "when the partner has housing payments" do
+      before do
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Partner").and_return(true)
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Applicant").and_return(false)
+      end
+
+      it { is_expected.to be :housing_benefits }
+    end
+
+    context "when there are no housing payments" do
+      before do
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Partner").and_return(false)
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Applicant").and_return(false)
+      end
+
+      it { is_expected.to be :has_dependants }
+    end
+  end
+
+  describe "#check_answers" do
+    subject { described_class.check_answers.call(legal_aid_application) }
+
+    context "when the applicant has housing payments and is uploading bank statements" do
+      it { is_expected.to be :housing_benefits }
+    end
+
+    context "when the partner has housing payments" do
+      before do
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Partner").and_return(true)
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Applicant").and_return(false)
+      end
+
+      it { is_expected.to be :housing_benefits }
+    end
+
+    context "when there are no housing payments" do
+      before do
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Partner").and_return(false)
+        allow(legal_aid_application).to receive(:housing_payments_for?).with("Applicant").and_return(false)
+      end
+
+      it { is_expected.to be :has_dependants }
+    end
+  end
+end
```

Comment:
There is quite a lot of duplication in the setup and allow blocks.
I used a slightly different pattern on spec/services/flow/steps/provider_capital/policy_disregards_step_spec.rb and wonder if it would work here too?

Reply:
Using `receive_messages`? Is it possible to do that and pass in an argument to the `housing_payments_for?` method? I can't seem to find a way of doing it